qiita_db/search.py

```diff
@@ -192,27 +192,31 @@ def __call__(self, searchstr, user):
 
         Metadata column names and string searches are case-sensitive
         """
-        study_sql, sample_sql, meta_headers = \
-            self._parse_study_search_string(searchstr, True)
-        conn_handler = SQLConnectionHandler()
-        # get all studies containing the metadata headers requested
-        study_ids = {x[0] for x in conn_handler.execute_fetchall(study_sql)}
-        # strip to only studies user has access to
-        if user.level not in {'admin', 'dev', 'superuser'}:
-            study_ids = study_ids.intersection(Study.get_by_status('public') |
-                                               user.user_studies |
-                                               user.shared_studies)
-
-        results = {}
-        # run search on each study to get out the matching samples
-        for sid in study_ids:
-            study_res = conn_handler.execute_fetchall(sample_sql.format(sid))
-            if study_res:
-                # only add study to results if actually has samples in results
-                results[sid] = study_res
-        self.results = results
-        self.meta_headers = meta_headers
-        return results, meta_headers
+        with TRN:
+            study_sql, sample_sql, meta_headers = \
+                self._parse_study_search_string(searchstr, True)
+
+            # get all studies containing the metadata headers requested
+            TRN.add(study_sql)
+            study_ids = set(TRN.execute_fetchflatten())
+            # strip to only studies user has access to
+            if user.level not in {'admin', 'dev', 'superuser'}:
+                study_ids = study_ids.intersection(
+                    Study.get_by_status('public') | user.user_studies |
+                    user.shared_studies)
+
+            results = {}
+            # run search on each study to get out the matching samples
+            for sid in study_ids:
+                TRN.add(sample_sql.format(sid))
+                study_res = TRN.execute_fetchindex()
+                if study_res:
+                    # only add study to results if actually has samples
+                    # in results
+                    results[sid] = study_res
+            self.results = results
+            self.meta_headers = meta_headers
+            return results, meta_headers
```

Comment:
It doesn't matter in this case. I've done it inside for consistency with other parts of the code, but it exits the context anyways...